lib/internal/filter.js

```diff
@@ -1,31 +1,19 @@
-import arrayMap from 'lodash/_arrayMap';
-import property from 'lodash/_baseProperty';
+import filter from 'lodash/filter';
 import noop from 'lodash/noop';
 import once from './once';
 
 export default function _filter(eachfn, arr, iteratee, callback) {
     callback = once(callback || noop);
-    var results = [];
+    var truthValues = new Array(arr.length);
     eachfn(arr, function (x, index, callback) {
         iteratee(x, function (err, v) {
-            if (err) {
-                callback(err);
-            }
-            else {
-                if (v) {
-                    results.push({index: index, value: x});
-                }
-                callback();
-            }
+            truthValues[index] = !!v;
+            callback(err);
         });
     }, function (err) {
-        if (err) {
-            callback(err);
-        }
-        else {
-            callback(null, arrayMap(results.sort(function (a, b) {
-                return a.index - b.index;
-            }), property('value')));
-        }
+        if (err) return callback(err);
+        callback(null, filter(arr, function (_, index) {
+            return truthValues[index];
```

Comment:
either unroll this as a simple loop `keys + for` loop or use a lodash object method and I'll approve